helix-core/src/main/java/org/apache/helix/cloud/azure/AzureCloudInstanceInformation.java

```diff
@@ -42,7 +42,8 @@ public String get(String key) {
   }
 
   public static class Builder {
-    private Map<String, String> _cloudInstanceInfoMap = null;
+
+    private final Map<String, String> _cloudInstanceInfoMap = new HashMap<>();
 
     public AzureCloudInstanceInformation build() {
       return new AzureCloudInstanceInformation(_cloudInstanceInfoMap);
```

Comment:
Safer to clone construct a new map so the newly built instances do not share anything.